lib/human_player.rb

```diff
@@ -0,0 +1,42 @@
+class HumanPlayer
+  attr_reader :display, :token
+
+  MESSAGES = {
+    player_prompt: "Go ",
+    select_space: "Please select your move by entering the number (1 - 9, from top left to bottom right) of an empty space.",
+    invalid_entry: "Invalid entry.",
+    space_taken: "Selection taken and not available.",
+  }
+
+  def initialize(display:, token:)
+    @display = display
+    @token = token
+  end
+
+  def selection(validator)
+    player_prompt
+    loop do
+      selection_prompt
+      selection = selection_valid(display.input, validator)
+      return selection if selection && selection_available(selection, validator)
+    end
```

Comment:
I like that you extracted the validator. I wonder if there is a way that we could leverage that more. Both of our players are still doing this looping logic. Is there some way we can encapsulate that? Perhaps we could push that down into the validator, or alternatively, make it the responsibility of the client of the player objects?

Reply:
I removed the looping logic from the computer player